libavfilter/vf_gblur_vulkan.c

```diff
@@ -0,0 +1,472 @@
+        GLSLC(0, void main()                                              );
+        GLSLC(0, {                                                        );
+        GLSLC(1,     ivec2 size;                                          );
+        GLSLC(1,     const ivec2 pos = ivec2(gl_GlobalInvocationID.xy);   );
+        for (int i = 0; i < planes; i++) {
+            GLSLC(0,                                                      );
+            GLSLF(1,  size = imageSize(output_image[%i]);              , i);
+            GLSLC(1,  if (IS_WITHIN(pos, size)) {                         );
+            GLSLF(2,      gblur(pos, %i);                              , i);
+            GLSLC(1, }                                                    );
+        }
+        GLSLC(0, }                                                        );
+
+        RET(ff_vk_compile_shader(ctx, shd, "main"));
+
+        RET(ff_vk_init_pipeline_layout(ctx, s->pl_hor));
+        RET(ff_vk_init_compute_pipeline(ctx, s->pl_hor));
+
+        ff_vk_update_descriptor_set(ctx, s->pl_hor, 1);
+    }
+
+    { /* Create shader for the vertical pass */
+        image_descs[0].updater = s->tmp_images;
+        image_descs[1].updater = s->output_images;
+
+        s->pl_ver = ff_vk_create_pipeline(ctx);
+        if (!s->pl_ver)
+            return AVERROR(ENOMEM);
+
+        shd = ff_vk_init_shader(ctx, s->pl_ver, "gblur_compute_ver", image_descs[0].stages);
+        ff_vk_set_compute_shader_sizes(ctx, shd, (int [3]){ CGS, CGS, 1 });
+
+        RET(ff_vk_add_descriptor_set(ctx, s->pl_ver, shd, image_descs, FF_ARRAY_ELEMS(image_descs), 0));
+        RET(ff_vk_add_descriptor_set(ctx, s->pl_ver, shd, &buf_desc, 1, 0));
+
+        GLSLD(   gblur_vertical                                           );
+        GLSLC(0, void main()                                              );
+        GLSLC(0, {                                                        );
+        GLSLC(1,     ivec2 size;                                          );
+        GLSLC(1,     const ivec2 pos = ivec2(gl_GlobalInvocationID.xy);   );
+        for (int i = 0; i < planes; i++) {
+            GLSLC(0,                                                      );
+            GLSLF(1,  size = imageSize(output_image[%i]);              , i);
+            GLSLC(1,  if (IS_WITHIN(pos, size)) {                         );
+            GLSLF(2,      gblur(pos, %i);                              , i);
+            GLSLC(1, }                                                    );
+        }
+        GLSLC(0, }                                                        );
+
+        RET(ff_vk_compile_shader(ctx, shd, "main"));
+
+        RET(ff_vk_init_pipeline_layout(ctx, s->pl_ver));
+        RET(ff_vk_init_compute_pipeline(ctx, s->pl_ver));
+
+        ff_vk_update_descriptor_set(ctx, s->pl_ver, 1);
+    }
+
+    RET(ff_vk_create_exec_ctx(ctx, &s->exec));
+
+    s->initialized = 1;
+    av_free(kernel_def);
+
+    return 0;
+fail:
```

Comment:
do we need free kernel_def?

Reply:
Sure. If not we may have a memory leak here. The memory chunk for kernel_def was allocated in the function, av_asprintf.